File: 1957. Delete Characters to Make Fancy String.cpp
class Solution {
public:
    string makeFancyString(string s) {
        int n = s.length();
        if (n < 3) return s;  // If the string has fewer than 3 characters, it's already fancy.

        string result = s.substr(0, 2);  // Initialize the result string with the first two characters.
        
        // Traverse the string starting from the third character.
        for (int i = 2; i < n; ++i) {
            // Check if the current character does not form three consecutive identical characters.
            if (s[i] != s[i - 1] || s[i] != s[i - 2]) {
                result += s[i];  // Append the current character to the result.
            }
        }
        
        return result;
    }
};

File: 1957. Delete Characters to Make Fancy String.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder result = new StringBuilder();
        for(int i=0;i<s.length();i++){
            
                if(i + 2 < s.length() && s.charAt(i) == s.charAt(i+1) && s.charAt(i+1) == s.charAt(i+2)){
                    continue;
                }
                else {
                    result.append(s.charAt(i));
                }
        }
        return result.toString();
    }
}
